### DISCUSSION TITLE
Welcome to node-red-contrib-join-wait Discussions!

### DISCUSSION THREAD

**Discussion by dxdc** [dxdc/node-red-contrib-join-wait#3] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by DavyJans** [comment]
Hi,

I'm using your node in a flow and it does help me achieve what I wanted (merge parallel data objects) I have the feeling it doesn't do it as designed or described in the manual.

I get a msg back that has a payload but this one only contains the first or last message. The messages are not really joined in here. There is however also the paths object that das path_1 and path_2 so that is enough for me to work with (I use it in a function to create another object type). 

Background: I work with 2 calls, 1 subscription to an mqtt topic and when that one returns a value (json) I do a call to openweather api. Then I want to "model" these 2 responses in 1 new object.

Is what I am seeing expected?

**Reply by dxdc** [reply]
@DavyJans I think you may be better off with a `join` node. The purpose of this node is that the 2 responses (or more...) also have the condition that they need to be received in a certain time interval. It doesn't sound like that last part is important in your case.

E.g. Await MQTT topic, then call Openweather API, then merge the results.

That being said, I'll be curious what settings you have. You can also try setting the 'Merged' option to 'overwrite' if you haven't tested that already.

If you can post your flow (substituting using an inject node for example) that would be helpful for analysis. Also, give something like the below a try. I think it's more in line with what you want, since there doesn't appear to be a time-component for you.

```
[{"id":"98e95eb4.f8464","type":"inject","z":"3efa8464.f9cb8c","name":"","props":[{"p":"payload"},{"p":"topic","vt":"str"}],"repeat":"","crontab":"","once":false,"onceDelay":0.1,"topic":"key_1","payload":"","payloadType":"date","x":430,"y":740,"wires":[["4a6bfd1a.949244","ffe71e4f.9a0e5"]]},{"id":"4a6bfd1a.949244","type":"delay","z":"3efa8464.f9cb8c","name":"","pauseType":"delay","timeout":"1","timeoutUnits":"seconds","rate":"1","nbRateUnits":"1","rateUnits":"second","randomFirst":"1","randomLast":"5","randomUnits":"seconds","drop":false,"x":520,"y":620,"wires":[["f8f2609c.31a81"]]},{"id":"ffe71e4f.9a0e5","type":"join","z":"3efa8464.f9cb8c","name":"","mode":"custom","build":"object","property":"payload","propertyType":"msg","key":"topic","joiner":"\\n","joinerType":"str","accumulate":false,"timeout":"","count":"2","reduceRight":false,"reduceExp":"","reduceInit":"","reduceInitType":"num","reduceFixup":"","x":770,"y":740,"wires":[["43cee567.5137ac"]]},{"id":"43cee567.5137ac","type":"debug","z":"3efa8464.f9cb8c","name":"","active":true,"tosidebar":true,"console":false,"tostatus":false,"complete":"true","targetType":"full","statusVal":"","statusType":"auto","x":930,"y":740,"wires":[]},{"id":"f8f2609c.31a81","type":"change","z":"3efa8464.f9cb8c","name":"","rules":[{"t":"set","p":"payload","pt":"msg","to":"","tot":"date"},{"t":"set","p":"topic","pt":"msg","to":"key_2","tot":"str"}],"action":"","property":"","from":"","to":"","reg":false,"x":700,"y":620,"wires":[["ffe71e4f.9a0e5"]]}]
```
